tests/unit/test_charm.py

```diff
@@ -547,3 +556,59 @@ def test_given_cmd_when_app_environment_variables_set_then_command_executed_with
                 "NO_PROXY": "No proxy",
             },
         )
+
+    @patch(
+        f"{CERT_TRANSFER_LIB_PATH}.CertificateTransferProvides.add_certificates",
+    )
+    def test_given_cert_transfer_relation_not_created_then_ca_certificates_not_added_in_relation_data(  # noqa: E501
```

Comment:
do we need to patch this? we can just look at the provider databag and see that its empty

Reply:
I'm not sure how to do that. If there is no relation, there is no databag, I can't do `harnsess.get_relation_data(relation_id, app_name)`.
I fixed it for the other comment however.

Reply:
in that case wouldn't the charm just raise an error? So we don't need to do either, we can just wait for the test to raise if it breaks.

Reply:
No, the charm will check whether the relation was created and it won't crash or raise.

Reply:
i mean if this feature was bugged and the charm attempted to add ca certificates to a relation that is not created. In that case nothing will happen?

Reply:
That's more on the lib to check, the charm will not directly interact with relation data.